checker.py

```diff
@@ -227,9 +225,27 @@ def git_checker(
         if not check_clean(lines):
             # WD dirty
             dirty_dirs += [gd]
-        if check_unpushed(lines):
-            # Changes unpushed
-            unpushed_dirs += [gd]
+
+    # Then, check for unpushed branches with a configured remote. This is done by using
+    # git log remote/branch..branch.
+    unpushed_branches: List[str] = []
+    for gd in itr:
+        # Retrieve all branches with a configured remote
+        branches_with_remote = ['git', 'config', '--get-regexp', '^branch\..*\.remote$']
+        p = sp.Popen(branches_with_remote, stdout=sp.PIPE, universal_newlines = True, cwd=gd)
+        res, ess = p.communicate()
+        branches_with_remotes = [(r.split('.')[1], r.split(' ')[1]) for r in res.splitlines()]
+
+        # Check each branch with a remote which is not pushed
+        for (branch, remote) in branches_with_remotes:
+            # Check which commits are on branch, but not on remote/branch
+            query = f"{remote}/{branch}..{branch}"
+            p = sp.Popen(['git', 'log', query], stdout=sp.PIPE, universal_newlines=True, cwd=gd)
+            res, ess = p.communicate()
+            # Find what's important.
+            if res != "":
+                # Changes unpushed
+                unpushed_branches += [f"{gd}, branch {branch}"]
```

Comment:
What do you think about omitting reporting the branch if it is `master` and it is the only one? This could keep the output simpler.

Reply:
Done with d09c54c.